### ISSUE TITLE
[java] SimplifyBooleanReturns: False positive for complex return values

### ISSUE THREAD

**Issue by uhafner** [pmd/pmd#6449]
**Affects PMD Version: 7.20.0**

**Rule:** [SimplifyBooleanReturns](https://docs.pmd-code.org/latest/pmd_rules_java_design.html#simplifybooleanreturns)

**Description:**

SimplifyBooleanReturns reports a violation even if the else branch is very complex statement. This would make the code more unreadable IMHO.

See also #5219, which seems to be related.

**Code Sample demonstrating the issue:**

```java
if (lastCompletedBuild == null) {
    return true;
}

return lastCompletedBuild.getActions(ResultAction.class)
        .stream()
        .filter(createToolFilter(selectTools, tools))
        .anyMatch(resultAction -> resultAction.getResult().getTotalSize() > 0);
}
```
Or 

```
if (getLineStart() <= line && line <= getLineEnd()) {
    return true; // the line is within the primary range of this issue
}
return getLocations().stream()
        .anyMatch(location -> location.contains(line));
```

**Expected outcome:**

PMD reports a violation, but I think that's too aggressive. That's a false positive IMHO. I think PMD should skip such else conditions.

**Running PMD through: Maven**
